### PR TITLE
Megapass on Chapter 13

### PR DESCRIPTION
**Pull request by pavpanchekha**
Hi Chris, sorry that this PR is late (I'm still getting the hang of fatherhood). This is mostly a language pass over Chapter 13. Once something like this is merged I'm ready to release chapter 13. Here are the major changes:

- Fix an issue wherein we used parent pointers earlier in the chapter than we introduced them
- Change the `CompositedLayer` API so that a composited layer always has at least one element (simplifies the code)
- Add extra text clarifying why we need identities on display items
- Update the overlap testing text to no longer refer to paint chunks

Also, a quick question:

- Do we need to make `transform` animatable? I reorganized that section a bit and realized that all of the animation code is now in one chunk at the end. I guess if it's not animatable, the reader might ask why we don't just set its `needs_compositing` to `False` (that wouldn't work, since the transform might contain a composited effect, but the reader might miss that) but we could clarify that. And implementing animated transforms sounds like a great exercise.

### CODE REVIEW

File: book/animations.md [hunk 1]
```diff
@@ -1859,23 +1818,23 @@ avoid slowing down transform animations.
 
 [cssfilter]: https://developer.mozilla.org/en-US/docs/Web/CSS/filter
 
-[^overlap-example]: In addition, it's not possible to create the overlapping
-squares example of this section without something like transforms. Real
-browsers have many other methods, such as [position]. In fact, it's very
-difficult to cause overlap at all in our current browser.
+[^overlap-example]: It's not possible to create the overlapping
+squares example of this section without something way of moving an
```
Comment (chrishtr) [inline book/animations.md:1822]:
```suggestion
squares example of this section without some way of moving an
```

File: book/animations.md [hunk 2]
```diff
@@ -1859,23 +1818,23 @@ avoid slowing down transform animations.
...
+element around. Real browsers have many methods for this, such as
+[position].
 
 [position]: https://developer.mozilla.org/en-US/docs/Web/CSS/position
 
-The `transform` CSS property in general lets you apply
-[any linear transform][transform-def] in 3D space, but I'll just cover really
-basic 2D translations. Here's HTML for the overlap example mentioned in the
-last section:[^why-zero]
+The `transform` CSS property in quite general and lets you apply [any
```
Comment (chrishtr) [inline book/animations.md:1828]:
```suggestion
The `transform` CSS property is quite powerful, and lets you apply [any
```

File: book/animations.md [hunk 3]
```diff
@@ -1887,8 +1846,8 @@ This particular rule is purely a historical artifact.
                 background-color:lightgreen;
                 transform:translate(0px, 0px)"></div>
 
-Adding in support for this kind of transform is not too hard: first
-just parse it:[^space-separated]
+Support these transform is simple. First let's parse the property
```
Comment (chrishtr) [inline book/animations.md:1849]:
```suggestion
Supporting these transforms is simple. First let's parse the property
```

File: book/animations.md [hunk 4]
```diff
@@ -2116,7 +2084,7 @@ without just giving up and compositing the scroller.
 Summary
 =======
 
-This chapter introduced the concept of animations. The key takeaways you should
+This chapter introduce animations. The key takeaways you should
```
Comment (chrishtr) [inline book/animations.md:2087]:
```suggestion
This chapter introduces animations. The key takeaways you should
```

File: src/lab13.py [hunk 5]
```diff
@@ -109,9 +109,6 @@ def needs_compositing(self):
         return any([child.needs_compositing() \
             for child in self.children])
 
-    def clone(self, children):
```
Comment (chrishtr) [inline src/lab13.py:112]: Why remove this?

**Comment by chrishtr** [comment]
> would be fine

On further reflection, I remembered additional reasons we should keep transform animations in the book:

* It's weird to have a chapter for animations where we get to the end and there is no animation that moves anything around
* Not too much work

However, I do think we can leave composited transform animations to an exercise. I am sending you a PR now for that.

File: book/animations.md [hunk 6]
```diff
@@ -351,10 +350,10 @@ the GPU. And as expected, speed is much improved:
     Time in render on average: 23ms
 
 That's about three times faster, almost fast enough to hit sixty
-frames per second. And on your computer, you'll likely see even more
-speedup than that, so perhaps your animations are already quite
-smooth. But if we want to go faster yet, we'll need to reduce the
-total amount of work in raster and draw.
+frames per second. (And on your computer, you'll likely see even more
+speedup than I did, so for you it might already be fast enough.) But
```
[the remark this answers is not in the document]
Comment (pavpanchekha) [reply book/animations.md:354]: Great point.

File: book/animations.md [hunk 7]
```diff
@@ -860,16 +856,12 @@ class Browser:
...
-Some problems remains in the way of great animation performance, though. The
-biggest is that we're doing *all* of these steps on every frame...but the whole
-point of compositing was to avoid re-rastering layers! Let's work on that
-next.
-
-[profiling]: http://localhost:8001/scheduling.html#profiling-rendering
+So simple and elegant! Now, on every frame, we are simply splitting
+the display list into composited layers and the draw display list, and
+then running each of those it their own phase. We're now half-way
+toward getting super-smooth animations. What remains is *not*
```
[the remark this answers is not in the document]
Comment (pavpanchekha) [reply book/animations.md:862]: Rephrased to make clearer.

File: book/animations.md [hunk 8]
```diff
@@ -1124,29 +1116,27 @@ The `style` function can animate any changed properties listed in
...
 
-Finally, we need to run the animations! Basically, every frame, we're
-going to want to find all the active animations on the page and call
-`animate` on them. Since these animations are a variation of
-JavaScript animations using `requestAnimationFrame`, let's run
-animations right after handling those callbacks:
+With the animations created, we now need to run them on every later
+frame. Basically, on every frame, we iterate through all the active
+animations on the page and call `animate` on them. Since these
+animations cousin to JavaScript using `requestAnimationFrame`, let's
```
[the remark this answers is not in the document]
Comment (pavpanchekha) [reply book/animations.md:1138]: Weird choice of word, rephrased.

File: src/lab13.py (hunk 5 above)
Comment (pavpanchekha) [reply src/lab13.py:112]: Oh—it doesn't matter either way. In Python it's just a runtime error to call an undefined method; adding `assert False` is only worth it if you write a better error message than the default, and we didn't. So I removed this code block from the chapter (I kept the text, just removed the code block) and then for symmetry removed it from the code as well.

**Pull request merged**